src/jdk.jshell/share/classes/jdk/jshell/SourceCodeAnalysisImpl.java

```diff
@@ -923,6 +984,20 @@ private boolean isVariable(TreePath tp) {
                tp.getParentPath().getLeaf().getKind() == Kind.VARIABLE;
     }
 
+    private boolean isAnnotationAttributeValue(TreePath tp) {
+        if (tp.getParentPath() == null) {
+            return false;
+        }
+        if (tp.getParentPath().getLeaf().getKind() == Kind.NEW_ARRAY &&
+            ((NewArrayTree) tp.getParentPath().getLeaf()).getInitializers().contains(tp.getLeaf())) {
+            return isAnnotationAttributeValue(tp.getParentPath());
```

Comment:
I recall JLS requires annotation arrays to be one-dimensional only, but this check seems to allow multi-dimensional arrays with annotation elements.